core/renderers/common/renderer.ts

```diff
@@ -257,6 +257,28 @@ export class Renderer implements IRegistrable {
     return InsertionMarkerManager.PREVIEW_TYPE.INSERTION_MARKER;
   }
 
+  /**
+   * Visually highlights the given connection with a border, if it is not
+   * already highlighted.
+   */
+  highlightConnection(conn: RenderedConnection): void {
+    const block = conn.getSourceBlock();
+
+    const info = this.makeRenderInfo_(block);
+    info.measure();
+    const drawer = this.makeDrawer_(block, info);
+
+    drawer.drawConnectionHighlightPath(conn);
+  }
+
+  /** Visually unhighlights the given connection, if it is highlighted. */
+  unhighlightConnection(conn: RenderedConnection): void {
+    const pathObject = conn.getSourceBlock().pathObject;
+    if (pathObject.removeConnectionHighlight) {
+      pathObject.removeConnectionHighlight(conn);
+    }
```

Comment:
```suggestion
   conn.getSourceBlock().pathObject.removeConnectionHighlight?.(conn);
```
(Assuming that `.getSourceBlock()` alwasy returns an object with a `.pathObject` property that is also an object, otherwise some more `.?`s are in order, though I imagine tsc would have told you already if so.)